src/components/Card.js

```diff
@@ -1,13 +1,42 @@
+import React from "react";
+
 import './Card.css'
 
-function Card() {
-  return (
-		<div className="card">
-			<h3 className="caption">Card Title</h3>
-			<hr></hr>
-			<p>Some Text</p>
-		</div>
-	);
+class Card extends React.Component {
+	constructor(props) {
+	  super(props)
+	  this.output = this.output.bind(this);
+	  this.state = {
+		divName: 'card'
+	  }
+	}
+	
+	output() {
+	  if(this.state.divName === 'card'){
+		this.setState({divName: 'card-item'});
+	  }
+	  if(this.state.divName === 'card-item'){
+		this.setState({divName: 'card'});
+	  }
+	  
+	}
+   
+	render() {
+	  return(
+		<div className={this.state.divName}>
+		 <div className={'card-inner__caption'}>
+		   <h3 className="card-caption">Card Title</h3>
+		   <input type="checkbox" onChange={this.output} />
+		 </div>
+  
+		 <hr></hr>
+  
+		 <div>
+		   <p>Some Text</p>
+		 </div>
+	   </div>
+	  )
```

Comment:
Adding a simple style change to the state itself is ambiguous. The thing to actually set and check in the state - whether card was 'checked' or not, use boolean value.
```
...
this.state = {
  checked: false,
};
...
```
Then you can use simple setState call as onChange event handler:
```
...
<input type="checkbox" onChange={()=> {this.setState((state) => ({ checked: !state.checked}))}} />
...
```
You can use simple ternary operator expression to define required className:
```
...
<div className={this.state.checked ? 'card' : 'card-item'}>
...
```

Also, to make code more readable and start using greater way to resolve conditional styling, you can use classnames package: https://www.npmjs.com/package/classnames